File: Python/power-of-four.py
class Solution:
    def isPowerOfFour1(self, num):
        """
        :type num: int
        :rtype: bool
        """
        x = 1

        while x != num:
            if x < num:
                x <<= 2
            else:
                return False

        return True
        
    def isPowerOfFour2(self, num):
        """
        :type num: int
        :rtype: bool
        """
        return num > 0 and (num & (num - 1)) == 0 and (num & 0b01010101010101010101010101010101) == num

File: Swift/power-of-four.swift
class Solution {
    func isPowerOfFour1(_ num: Int) -> Bool {
        var x = 1

        while x != num {
            if x < num {
                x <<= 2
            } else {
                return false
            }
        }

        return true
    }

    func isPowerOfFour2(_ num: Int) -> Bool {
        return num > 0 && (num & (num - 1)) == 0 && (num & 0b01010101010101010101010101010101) == num
    }
}
